modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/management_bar/components/ActiveViewSelector.js

```diff
@@ -16,10 +16,16 @@ import {ClayButtonWithIcon} from '@clayui/button';
 import ClayDropDown from '@clayui/drop-down';
 import ClayIcon from '@clayui/icon';
 import PropTypes from 'prop-types';
-import React, {useState} from 'react';
+import React, {useContext, useState} from 'react';
 
-function ActiveViewSelector({activeView, setActiveView, views}) {
+import DatasetDisplayContext from '../../DatasetDisplayContext';
+import persistActiveView from '../../thunks/persistActiveView';
+import ViewsContext from '../../views/ViewsContext';
+
+function ActiveViewSelector({views}) {
 	const [active, setActive] = useState(false);
+	const [{activeView}, dispatch] = useContext(ViewsContext);
+	const {id} = useContext(DatasetDisplayContext);
```

Comment:
So... how's id being generated and from where? Are we certain it is unique?

Just thinking that we have many `searchContainers` called `entriesSearchContainer` or something like that, so we need to provide a solid id'ing mechanism. We can leave it to the developers, but it seems like a big foot gun because the relation between id and persistence might not be obvious.

Reply:
This `id` is given from the developer at the time of the call to the TagLib. And for DataSet Display it's always been [mandatory](https://github.com/liferay/liferay-portal-ee/blob/7.1.x/modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/liferay-commerce-ui.tld#L142)

Reply:
Sorry, forgot to answer here... the thing is, being mandatory doesn't make it unique. We have several repeated id's for `liferay-ui:searchContainer`. Main offenders off the top of my head would be `entries` and maybe `assetEntries`.

So, my worry is that they might clash. Namespacing won't be enough on itself either, since `_foo_entries` might be different searchcontainers within the same app. Thus, maybe use the `servletContext` or something like that to make it more unique?

Reply:
Hum I see. I think either servletContext or portlet namespace end up being almost the same difference. The former might be more granular if a portlet is spread across multiple modules and those different modules call the DataSet Display taglib. In portal I see we tend to have multiple modules for java stuff but more often just one web module.

Reply:
See https://github.com/liferay-frontend/liferay-portal/pull/125/commits/08ab79f485a0f16797c4d105dcef8eda5aebee7a